### PR TITLE
Support `expand_dims`

### PR DESCRIPTION
**Pull request by asi1024**
Fixes https://github.com/pytorch/pytorch/issues/56774

This PR implements `torch.expand_dims` for the compatibility with NumPy’s interface.
cc: @mruberry, @rgommers, @emcastillo and @kmaehashi

### CODE REVIEW

**Review by mruberry — APPROVED** [review]
Cool! Thanks @asi1024! It's nice to see how much easier adding an alias has become, at least for simple cases.

I have just one question about whether we should alias the inplace version of the op, too. Might as well? 

cc @ngimel

File: aten/src/ATen/native/native_functions.yaml [hunk 1]
```diff
@@ -3772,6 +3772,10 @@
   dispatch:
     CompositeExplicitAutograd: unsqueeze_
 
+# expand_dims, alias of unsqueeze
+- func: expand_dims(Tensor(a) self, int dim) -> Tensor(a)
```
Comment (mruberry) [inline aten/src/ATen/native/native_functions.yaml:3776]: What's your thinking on aliasing the inplace version of the operation, unsqueeze_, as well?

Comment (asi1024) [reply aten/src/ATen/native/native_functions.yaml:3776]: The in-place version is out of Array API standard, so it does not necessarily need to be supported, but I think it's OK to support it. Done in 83a21ff!

Comment (mruberry) [reply aten/src/ATen/native/native_functions.yaml:3776]: Looks great; I made one small tweak to the inplace method docs. I think this is now ready to merge!

**Comment by mruberry** [comment]
Hey @asi1024, just fyi I'm going to need to follow-up on one internal hitch with this PR, so it may not land immediately

**Comment by asi1024** [comment]
@mruberry Thank you! Please let me know if there is any updates :smiley:

**Comment by mruberry** [comment]
> @mruberry Thank you! Please let me know if there is any updates 😃

I will; we're preparing for the PyTorch 1.9 release now, however, so it might take some time to follow-up, unfortunately.

**Comment by asi1024** [comment]
@mruberry Kindly ping :smiley:

**Comment by mruberry** [comment]
> @mruberry Kindly ping 😃

Thanks for the ping @asi1024. I'm hoping to follow-up on this soon.

**Comment by mruberry** [comment]
> > @mruberry Kindly ping 😃
> 
> Thanks for the ping @asi1024. I'm hoping to follow-up on this soon.

Still haven't forgotten about this. Just trying to work around the PyTorch 1.10 branch cut.

**Comment by mruberry** [comment]
> > Hey @asi1024, just fyi I'm going to need to follow-up on one internal hitch with this PR, so it may not land immediately
> 
> Details, from [#58742 (comment)](https://github.com/pytorch/pytorch/issues/58742#issuecomment-881588847): _Adding the alias would be great! `expand_dims` is on hold just because a dependency of PyTorch also defines an `expand_dims` function in a way where they can't build with PyTorch if it's added, and we're working with them to resolve this issue. This complication is atypical and unlikely to affect other aliases._
> 
> I assume this is a non-public package. Will it be possible to get them to fix it before the 1.11 release @mruberry?

Hard to know the timeline but I certainly hope so. I'm reviewing these internal failures today so hope to know more soon.

**Pull request closed without merging**